src/app.js

```diff
@@ -0,0 +1,124 @@
+const headline = (id, title, webUrl, apiUrl) => {
+  return {
+    id,
+    title,
+    webUrl,
+    apiUrl
+  }
+};
+
+class App {
+  constructor(makeHeadlinesRequest, makeSummaryRequest, makeArticleRequest) {
+    this.makeHeadlinesRequest = makeHeadlinesRequest;
+    this.makeSummaryRequest = makeSummaryRequest;
+    this.makeArticleRequest = makeArticleRequest;
+  }
+
+  getHeadlines () {
+    this.makeHeadlinesRequest(data => {
+      let idx = 0;
+      data.response.results.forEach((item) => {
+        idx++;
+        const newHeadline = headline (idx, item.webTitle, item.webUrl, item.apiUrl)
+        this._articleHeadline(newHeadline);
+      });
+    });
+  }
+
+  _getArticleImage = (articleDiv, articleUrl, idx) => {
+    this.makeArticleRequest(articleUrl, data => {
+      const doc = new DOMParser().parseFromString(data.response.content.fields.body, "text/html");
+      let imgUrl;
+      try {
+        imgUrl =  doc.querySelector("img").getAttribute("src");
+      } catch {
+        imgUrl = "./public/images/news.png";
+      }
+      this._articleImage(articleDiv, idx, imgUrl);
+    });
+  }
+
+  _getSummary = (caller, idx, articleUrl) => {
+    this.makeSummaryRequest(articleUrl, data => {
+      this._articleSummary(caller, idx, data.sentences, articleUrl);
+    });
+  }
+
+  _articleImage(articleDiv, idx, imgUrl) {
+    const img = document.createElement("img");
+    img.id = `image-${idx}`;
+    img.src = imgUrl;
+    articleDiv.prepend(img);
+  }
+
+  _articleHeadline (newHeadline) {
+    let articleDiv = this._articleArea(newHeadline.id);
+    this._getArticleImage(articleDiv, newHeadline.apiUrl, newHeadline.id);
+    this._articleTitle(articleDiv, newHeadline.id, newHeadline.title, newHeadline.webUrl);
+  }
+
```

Comment:
I'm not entirely sure what articleDiv means but generally your methods are easy to understand. Maybe you are using some technical terms which are more difficult to understand as well but I'm sure it wouldn't be an issue in a technical environment